### DISCUSSION TITLE
Is it possible to combine system prompt with vector database and document reader?

### DISCUSSION THREAD

**Discussion by rasmus-puls** [FlowiseAI/Flowise#1273] [Q&A] [answered]
Hello, I have made 2 different setups, but I need a combined one. I just cannot figure out how to do it - or if it is possible at all.

In the first setup I have provided a PDF document with some information that my chat bot must be able to answer questions about:
![image](https://github.com/FlowiseAI/Flowise/assets/68005043/0af8f41e-1327-4514-818e-217790f1c096)

In the second setup, I have configured a system prompt, instructing the chat bot with a personality and a scope of what it is allowed to asnwer questions about:
![image](https://github.com/FlowiseAI/Flowise/assets/68005043/ad107472-c8dd-4f1d-b9f9-783bd171ede4)

I really need a way to combine these two setups, it's important that my chat bot knows who it is and has access to the specialized information that GPT-4 doesn't know.

Is this possible?

Thanks 🙏

**Comment by rasmus-puls** [comment]
👋 It's very quiet in here 🙂
I thought this would be a super basic use case? I can't be the first who needs to define a system prompt and use just 1 other capability at the same time no? 🤔

**Reply by joaquin-torroba** [reply]
I do this with an agent + rag_tool.
- Agent: I can define the system.
- Rag_tool: Vectorization.

**Comment by HenryHengZJ** [comment, marked as answer]
@rasmus-puls take a look at prompt chaining with vector store template, it has what you need: 
![image](https://github.com/FlowiseAI/Flowise/assets/26460777/e55c3272-d71e-4fd8-b853-a229d770f48f)

**Reply by rasmus-puls** [reply]
Thanks!

**Comment by ildefonsorobinjohn** [comment]
Hello, is it possible to check in the vector or validate the PII data before proceeding with the upsert to ensure it will not be included? Thank you
